Tools/LyTestTools/ly_test_tools/benchmark/data_aggregator.py

```diff
@@ -0,0 +1,163 @@
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+from argparse import ArgumentParser
+import json
+from pathlib import Path
+import time
+import subprocess
+
+from ly_test_tools.mars.filebeat_client import FilebeatClient
+
+class BenchmarkPathException(Exception):
+    """Custom Exception class for invalid benchmark file paths."""
+    pass
+
+class BenchmarkDataAggregator(object):
+    def __init__(self, workspace, logger, test_suite):
+        self.build_dir = workspace.paths.build_directory()
+        self.results_dir = Path(workspace.paths.project(), 'user/Scripts/PerformanceBenchmarks')
+        self.test_suite = test_suite
+        self.filebeat_client = FilebeatClient(logger)
+
+    def _update_pass(self, pass_stats, entry):
+        '''
+        Modifies pass_stats dict keyed by pass name with the time recorded in a pass timestamp entry.
+
+        :param pass_stats: dict aggregating statistics from each pass (key: pass name, value: dict with stats)
+        :param entry: dict representing the timestamp entry of a pass
+        :return: Time (in nanoseconds) recorded by this pass
+        '''
+        name = entry['passName']
+        time_ns = entry['timestampResultInNanoseconds']
+        pass_entry = pass_stats.get(name, { 'totalTime': 0, 'maxTime': 0 })
+
+        pass_entry['maxTime'] = max(time_ns, pass_entry['maxTime'])
+        pass_entry['totalTime'] += time_ns
+        pass_stats[name] = pass_entry
+        return time_ns
+
+
+    def _process_benchmark(self, benchmark_dir, benchmark_metadata):
+        '''
+        Aggregates data from results from a single benchmark contained in a subdirectory of self.results_dir.
+
+        :param benchmark_dir: Path of directory containing the benchmark results
+        :param benchmark_metadata: Dict with benchmark metadata mutated with additional info from metadata file
+        :return: Tuple with two indexes:
+            [0]: Dict aggregating statistics from frame times (key: stat name)
+            [1]: Dict aggregating statistics from pass times (key: pass name, value: dict with stats)
+        '''
+        # Parse benchmark metadata
+        metadata_file = benchmark_dir / 'benchmark_metadata.json'
+        if metadata_file.exists():
+            data = json.loads(metadata_file.read_text())
+            benchmark_metadata.update(data['ClassData'])
+        else:
+            raise BenchmarkPathException(f'Metadata file could not be found at {metadata_file}')
+
+        # data structures aggregating statistics from timestamp logs
+        frame_stats = { 'count': 0, 'totalTime': 0, 'maxTime': 0, 'minTime': float('inf') }
+        pass_stats = {}  # key: pass name, value: dict with totalTime and maxTime keys
+
+        # this allows us to add additional data if necessary, e.g. frame_test_timestamps.json
+        is_timestamp_file = lambda file: file.name.startswith('frame') and file.name.endswith('_timestamps.json')
```

Comment:
nit- do we care about casing here?

Reply:
Most of these files are autogenerated with the `frame` prefix and `_timestamps` suffix. I don't know if we want to add casing support but a simple` .lower()` call will do if you want.